### PR TITLE
refactor(date): updated errorMessage story and modified to import props

### PR DESCRIPTION
**Pull request by ryo-manba**
Closes # 

## 📝 Description

Modified to import HelpTextProps and ValidationResult from `@react-types/shared`.
Fixed the storybook to display `errorMessage`. Added a pattern to the docs for passing `errorMessage` as a function.

## ⛳️ Current behavior (updates)

> Please describe the current behavior that you are modifying

## 🚀 New behavior

> Please describe the behavior or changes this PR adds

## Summary by CodeRabbit

- **New Features**
  - Introduced `DateRangePicker` component with error message functionality for stay duration input validation.
  - Added `DateInput` component with error message functionality for date validation.

- **Enhancements**
  - Updated `time-input` stories to include error message configurations with `WithErrorMessage` and `WithErrorMessageFunction` exports.

### CODE REVIEW

**Comment by jrgarciadev** [comment]
@ryo-manba please add these examples to the date input docs as well and verify the time-input one

**Comment by jrgarciadev** [comment]
@ryo-manba add changeset

**Comment by ryo-manba** [comment]
We might want to add the pattern of passing `errorMessage` as a function to the docs for other components as well, but I will address that in a separate PR.

**Comment by ryo-manba** [comment]
@jrgarciadev Fixed it!

File: apps/docs/content/components/date-range-picker/error-message-function.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+import {parseDate} from "@internationalized/date";
+
+export default function App() {
+  return (
+    <DateRangePicker 
+      isInvalid
+      label="Stay duration" 
+      variant="bordered"
+      errorMessage="Please enter your stay duration"
+      errorMessage={(value) => {
```
Comment (wingkwong) [inline apps/docs/content/components/date-range-picker/error-message-function.ts:11]: may double check if two `errorMessage` are required or not

File: packages/components/date-input/stories/time-input.stories.tsx [hunk 2]
```diff
@@ -192,6 +193,31 @@ export const WithDescription = {
...
+
+export const WithErrorMessageFunction = {
+  render: Template,
+
+  args: {
+    ...defaultProps,
+    isInvalid: true,
+    errorMessage: "Please enter a valid time",
+  },
+  errorMessage: (value: ValidationResult) => {
```
Comment (wingkwong) [inline packages/components/date-input/stories/time-input.stories.tsx:214]: may double check if two `errorMessage` are required or not

Comment (ryo-manba) [reply packages/components/date-input/stories/time-input.stories.tsx]: Thanks I missed it.

File: apps/docs/content/components/date-input/error-message-function.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+import {CalendarDate, parseDate} from "@internationalized/date";
+
+export default function App() {
+  return (
+    <div className="flex w-full flex-wrap md:flex-nowrap gap-4">
+      <DateInput 
+        label={"Birth date"} 
+        defaultValue={parseDate("2024-04-04")} 
+        placeholderValue={new CalendarDate(1995, 11, 6)} 
+        description={"Thiis is my birth date."}
```
Comment (wingkwong) [inline apps/docs/content/components/date-input/error-message-function.ts:11]: typo: `Thiis`

**Review by wingkwong — APPROVED** [review]
One more minor change.

File: apps/docs/content/components/date-input/error-message-function.ts (hunk 3 above)
Comment (ryo-manba) [inline apps/docs/content/components/date-input/error-message-function.ts:11]:
```suggestion
        description={"This is my birth date."}
```

**Review by jrgarciadev — APPROVED** [review]

**Pull request merged**